src/plan/aggregate.rs

```diff
@@ -102,68 +104,61 @@ impl<P: Implementable> Implementable for Aggregate<P> {
         // resulting collections, s.t. they can be joined afterwards.
         for (i, aggregation_fn) in self.aggregation_fns.iter().enumerate() {
             let value_offset = value_offsets[i];
+            let with_lenght = self.with_symbols.len();
+
+            // Access the right value for the given iteration loop and extend possible with-values.
+            let prepare_unary = move |(key, tuple): (Vec<Value>, Vec<Value>)| {
```

Comment:
Didn't we used to have a prepare_unary? Where did that go in the meantime?

Reply:
Yeah I removed it the merge before, as every aggregation matched the type `Value` for its specific  computation first. But now we need to keep `Values` and not the primitive types first, s.t. we can append the specified `:find` values into it. Now the aggregations match their value later